src/test/ui/closures/closure-bounds-subtype.stderr

```diff
@@ -11,6 +11,10 @@ note: required by a bound in `take_const_owned`
    |
 LL | fn take_const_owned<F>(_: F) where F: FnOnce() + Sync + Send {
    |                                                  ^^^^ required by this bound in `take_const_owned`
+help: use parentheses to call this type parameter
+   |
+LL |     take_const_owned(f());
```

Comment:
This is, unfortunately, an erroneous suggestion that is uncovered by the generalizations I made in the last commit in the stack. However, this is not really the fault of the changes I made. For example, a similarly erroneous suggestion already triggers on nightly with this code:

```rust
struct Foo;

fn foo() -> Foo {
    todo!()
}

trait Bar {}

impl Bar for Foo {}

fn needs_foo<T>(_: impl Bar + Fn() -> T) { }

fn main() {
    needs_foo(foo);
}
```

Results in:
```
error[E0277]: the trait bound `fn() -> Foo {foo}: Bar` is not satisfied
  --> src/main.rs:14:15
   |
3  | fn foo() -> Foo {
   |    --- consider calling this function
...
14 |     needs_foo(foo);
   |     --------- ^^^ the trait `Bar` is not implemented for fn item `fn() -> Foo {foo}`
   |     |
   |     required by a bound introduced by this call
   |
note: required by a bound in `needs_foo`
  --> src/main.rs:11:25
   |
11 | fn needs_foo<T>(_: impl Bar + Fn() -> T) { }
   |                         ^^^ required by this bound in `needs_foo`
help: use parentheses to call the function
   |
14 |     needs_foo(foo());
   |                  ++
```

But adding those parentheses causes:
```
error[E0277]: expected a `Fn<()>` closure, found `Foo`
  --> src/main.rs:14:15
   |
14 |     needs_foo(foo());
   |     --------- ^^^^^ expected an `Fn<()>` closure, found `Foo`
   |     |
   |     required by a bound introduced by this call
   |
   = help: the trait `Fn<()>` is not implemented for `Foo`
   = note: wrap the `Foo` in a closure with no arguments: `|| { /* code */ }`
note: required by a bound in `needs_foo`
  --> src/main.rs:11:31
   |
11 | fn needs_foo<T>(_: impl Bar + Fn() -> T) { }
   |                               ^^^^^^^^^ required by this bound in `needs_foo`

For more information about this error, try `rustc --explain E0277`.
```

This is a fundamental limitation to the suggestion being implemented here in trait selection -- I'm happy to brainstorm ways of suppressing this, but I'm happy with keeping it as is and filing a follow-up bug.